### DISCUSSION TITLE
Not really happy with the rendered HTML of button

### DISCUSSION THREAD

**Discussion by vipindigiqt** [mantinedev/mantine#1428] [Feedback]
I recently found mantine, and I am just in love.

I really like the available components and a wide list of hooks.

I was just playing with the `Button` component the other day, and found out the rendered HTML like this.

![image](https://user-images.githubusercontent.com/81612126/167785044-6cb81a53-efe8-4148-8e28-a17328ffc9a8.png)

`<button class="mantine-Button-filled mantine-Button-root mantine-Group-child mantine-5q9waz" type="button">
   <div class="mantine-3xbgk5 mantine-Button-inner"><span class="mantine-qo1k2 mantine-Button-label">Settings</span></div>
</button>`

Just curious, why do we need an entire `<div>` inside a button?

A `span`, or any inline HTML is fine, but I don't think we should have these block level elements inside a `button`.

**Comment by rtivital** [comment]
What is wrong with using div inside button?

**Reply by vipindigiqt** [reply]
I don't think it's semantically correct.

https://www.w3.org/TR/2012/WD-html5-20120329/the-button-element.html#the-button-element

The w3 document says that It should only contain [Phrasing content](https://www.w3.org/TR/2012/WD-html5-20120329/content-models.html#phrasing-content).
